### DISCUSSION TITLE
Next.js

### DISCUSSION THREAD

**Discussion by CodeRider101** [SE-TINF22B2/G2-Schoolify#58] [Technologien]
Warum benutzen wir Next.js und nicht andere Technologien auf dieser Ebene?
@two-Gee @laralngn

**Comment by two-Gee** [comment]
### Erklärung
Next.js ist ein Full-Stack React Framework. Dabei stellt es fast alle Erweiterungen und Features, um ein performantes Frontend zu Verfügung zu stellen. Die wichtigsten dieser Erweiterung sind: 
- Eingebaute Route Handler 
- Eingebaute Optimierung für Bilder und Skripte
- Server-Side Rendering 
- Gute Developer Experience und übersichtliche Struktur 

### Warum haben wir uns für Next.js entschieden?
- Einige aus unserem Team kennen sich schon mit React aus und haben damit bereits entwickelt, Next.js besitzt einen intuitiven Aufbau und hat nur einen **geringen Lernaufwand**, wenn man schon React kennt. 
- Der eingebaute Route Handler ist für uns wichtig, da wir mehrere Seiten und Unterseiten benötigen und wir mit dem vorgegeben Aufbau von Next.js bereits **fertige Routen** als auch einen **übersichtlichen Aufbau** des Programmcodes haben. 
- Viele Seiten oder Komponenten unseres Schulplaners sind schon serverseitig renderbar, da sie sich nicht ständig ändern und oft nur Informationen anzeigen. (aktuelle Neuigkeiten, Stundenplan, Mensaplan) Mit Serverseitigen Rendering kann so eine deutlich bessere Performance erzielt werden.
